common/shell/shell_platform.c

```diff
@@ -0,0 +1,607 @@
+    for (int sen_idx=0; sen_idx<SDR_NUM; sen_idx++)
+        sensor_access(shell, sensor_config[sen_idx].num, SENSOR_READ);
+
+    shell_print(shell, "---------------------------------------------------------------------------------");
+}
+
+static void cmd_sensor_cfg_get(const struct shell *shell, size_t argc, char **argv)
+{
+    if (argc != 2) {
+        shell_warn(shell, "Try: platform sensor get <sensor_num>");
+        return;
+    }
+
+    int sen_num = strtol(argv[1], NULL, 16);
+
+    sensor_access(shell, sen_num, SENSOR_READ);
+}
+
+static void cmd_sensor_cfg_set_polling(const struct shell *shell, size_t argc, char **argv)
+{
+    if (argc != 3) {
+        shell_warn(shell, "Try: platform sensor set polling <sensor_num> <enable/disable>");
+        return;
+    }
+    /* TODO */
+}
+
+static void cmd_sensor_cfg_set_mbr(const struct shell *shell, size_t argc, char **argv)
+{
+    shell_warn(shell, "Set sensor MBR is not support!");
+    /* TODO */
+}
+
+static void cmd_sensor_cfg_set_threshold(const struct shell *shell, size_t argc, char **argv)
+{
+    shell_warn(shell, "Set sensor THRESHOLD is not support!");
+    /* TODO */
+}
+#endif
+
+/* 
+    Command PORT
+*/
+#if COMMAND_PORT == 1
+static void cmd_port_monitor_all(const struct shell *shell, size_t argc, char **argv)
+{
+    if (argc != 1) {
+        shell_warn(shell, "Try: platform port monitor_all");
+        return;
+    }
+}
+#endif
+
+/* 
+    Command I2C SLAVE
+*/
+//TODO
+
+/*********************************************************************************************************
+ * COMMAND DECLARE SECTION
+**********************************************************************************************************/
+
+/* GPIO sub command */
+#if COMMAND_GPIO == 1
+static void device_gpio_name_get(size_t idx, struct shell_static_entry *entry)
+{
+    const struct device *dev = shell_device_lookup(idx, GPIO_DEVICE_PREFIX);
+
+    entry->syntax = (dev != NULL) ? dev->name : NULL;
+    entry->handler = NULL;
+    entry->help = NULL;
+    entry->subcmd = NULL;
+}
+SHELL_DYNAMIC_CMD_CREATE(gpio_device_name, device_gpio_name_get);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_gpio_set_cmds,
+    SHELL_CMD(val, NULL, "Set pin value.", cmd_gpio_cfg_set_val),
+    SHELL_CMD(dir, NULL, "Set pin direction.", cmd_gpio_cfg_set_dir),
+    SHELL_CMD(int_type, NULL, "Set pin type of interrupt.", cmd_gpio_cfg_set_int_type),
+    SHELL_SUBCMD_SET_END
+);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_gpio_cmds,
+    SHELL_CMD(list_group, &gpio_device_name, "List all GPIO config from certain group.", cmd_gpio_cfg_list_group),
+    SHELL_CMD(list_all, NULL, "List all GPIO config.", cmd_gpio_cfg_list_all),
+    SHELL_CMD(get, NULL, "Get GPIO config", cmd_gpio_cfg_get),
+    SHELL_CMD(set, &sub_gpio_set_cmds, "Set certain GPIO config", NULL),
+    SHELL_CMD(multifnctl, NULL, "List all GPIO multi-function control regs list.", cmd_gpio_muti_fn_ctl_list),
```

Comment:
Remove last 'list' as it is redundant.

Reply:
Got it, this suggestion would add in next pr, thanks!